File: examples/indexer/javascript/SearchTransactionsNote.js
// SearchTransactionsNote.js
// requires algosdk@1.6.1 or higher 
// verify installed version
// npm list algosdk

const algosdk = require('algosdk');
const indexer_token = "";
const indexer_server = "http://localhost";
const indexer_port = 8980;

// Instantiate the indexer client wrapper
let indexerClient = new algosdk.Indexer(indexer_token, indexer_server, indexer_port);

(async () => {
    //let s = buffer.toString('base64');   
    let s = "c2hvd2luZyBwcmVmaXg=";
    let transactionInfo = await indexerClient.searchForTransactions()
        .notePrefix(s).do();
    console.log("Information for Transaction search: " + JSON.stringify(transactionInfo, undefined, 2));
})().catch(e => {
    console.log(e);
    console.trace();
});

File: examples/indexer/python/search_transactions_note.py
# search_transactions_note.py
import base64
import json
# requires Python SDK version 1.3 or higher
from algosdk.v2client import indexer

# instantiate indexer client
myindexer = indexer.IndexerClient(indexer_token="", indexer_address="http://localhost:8980")

import base64

note_prefix = 'showing prefix'.encode()

response = myindexer.search_transactions(
    note_prefix=note_prefix)

print("note_prefix = " +
      json.dumps(response, indent=2, sort_keys=True))
